paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/ContinuousFileSplitEnumerator.java

```diff
@@ -205,9 +210,13 @@ private Map<Integer, List<FileStoreSourceSplit>> createAssignment() {
                                 readersAwaitingSplit.remove(task);
                                 return;
                             }
+                            List<FileStoreSourceSplit> taskAssignment = new ArrayList<>();
+                            while (!splits.isEmpty() && taskAssignment.size() < splitBatchSize) {
```

Comment:
I think it is OK to just assign one split to a task for a bucket.
Maybe here we can just limit assignment batch.
```
List<FileStoreSourceSplit> taskAssignment = assignment.computeIfAbsent(task, i -> new ArrayList<>());
if (taskAssignment.size() < splitBatchSize) {
   taskAssignment.add(splits.poll());
}
```

This can be limited to a large number of buckets, and only a few tasks can work properly.
For example, buckets = 1000, but there is only one task.